### PR TITLE
cs/hardcoded-credentials - Removes false positive matches on benign Microsoft.AspNetCore.Identity properties

### PR DESCRIPTION
**Pull request by felickz**
Resolves [cs/hardcoded-credentials](https://github.com/github/codeql/blob/main/csharp/ql/src/Security%20Features/CWE-798/HardcodedCredentials.ql) false positive scenarios
- [UserOptions.AllowedUserNameCharacters](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.useroptions.allowedusernamecharacters?view=aspnetcore-7.0)
   - ex:  [WebGoat.NET](https://github.com/tobyash86/WebGoat.NET/blob/master/WebGoat.NET/Startup.cs#L90)
   ```csharp
    // User settings.
    options.User.AllowedUserNameCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+";
    ```
  - ex: OrchardCore
     - https://github.com/OrchardCMS/OrchardCore/blob/83ada8fbb11e208caab7d3111b33deb056009d1a/src/OrchardCore.Modules/OrchardCore.Setup/Startup.cs#L44
     - https://github.com/OrchardCMS/OrchardCore/blob/83ada8fbb11e208caab7d3111b33deb056009d1a/src/OrchardCore.Modules/OrchardCore.Users/Startup.cs#L166
     - https://github.com/OrchardCMS/OrchardCore/blob/83ada8fbb11e208caab7d3111b33deb056009d1a/test/OrchardCore.Tests/OrchardCore.Users/UsersMockHelper.cs#L9C1-L10C1
- [ClaimsIdentityOptions.UserNameClaimType](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.claimsidentityoptions.usernameclaimtype?view=aspnetcore-7.0#microsoft-aspnetcore-identity-claimsidentityoptions-usernameclaimtype)
   - ex:  [Multitenant.Auth](https://github.com/testing-felickz/MultiTenantAuth/blob/da70256e63df51d4ad838f3ea49c62c02c38aaec/MultiTenant.Auth/Extensions/AspIdentity/AccountOptions.cs#L33-L33)
   ```csharp
    options.ClaimsIdentity.UserNameClaimType = "username";
   ```

<details><summary>Test results before query update</summary>
<p>

```diff
failed: / workspaces / codeql / csharp / ql / test / query - tests / Security Features / CWE - 798 / HardcodedCredentials.qlref
-- - expected
++ + actual
@@ -9,6 + 9,8 @@
 | HardcodedCredentials.cs:52:13:52:24 | "myPa55word" | semmle.label | "myPa55word" |
 | HardcodedCredentials.cs:75:31:75:42 | "myusername" | semmle.label | "myusername" |
 | HardcodedCredentials.cs:75:45:75:56 | "mypassword" | semmle.label | "mypassword" |
+| HardcodedCredentials.cs:82:45:82:113 | "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+" | semmle.label | "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+" |
+| HardcodedCredentials.cs:89:33:89:42 | "username" | semmle.label | "username" |
 | TestHardcodedCredentials.cs:21:31:21:42 | "myusername" | semmle.label | "myusername" |
 | TestHardcodedCredentials.cs:21:45:21:56 | "mypassword" | semmle.label | "mypassword" |
 | TestHardcodedCredentials.cs:26:19:26:28 | "username" | semmle.label | "username" |
@@ -21,4 + 23,6 @@
 | HardcodedCredentials.cs:52:13:52:24 | "myPa55word" | HardcodedCredentials.cs:52:13:52:24 | "myPa55word" | HardcodedCredentials.cs:52:13:52:24 | "myPa55word" | The hard - coded value "myPa55word" flows to the $@ parameter in $@. | HardcodedCredentials.cs:52:13:52:24 | "myPa55word" | password | HardcodedCredentials.cs:50:33:52:25 | object creation of type X509Certificate2 | object creation of type X509Certificate2 |
 | HardcodedCredentials.cs:75:31:75:42 | "myusername" | HardcodedCredentials.cs:75:31:75:42 | "myusername" | HardcodedCredentials.cs:75:31:75:42 | "myusername" | The hard - coded value "myusername" flows to the $@ parameter in $@. | HardcodedCredentials.cs:75:31:75:42 | "myusername" | username | HardcodedCredentials.cs:75:9:75:57 | call to method CreateUser | call to method CreateUser |
 | HardcodedCredentials.cs:75:45:75:56 | "mypassword" | HardcodedCredentials.cs:75:45:75:56 | "mypassword" | HardcodedCredentials.cs:75:45:75:56 | "mypassword" | The hard - coded value "mypassword" flows to the $@ parameter in $@. | HardcodedCredentials.cs:75:45:75:56 | "mypassword" | password | HardcodedCredentials.cs:75:9:75:57 | call to method CreateUser | call to method CreateUser |
+| HardcodedCredentials.cs:82:45:82:113 | "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+" | HardcodedCredentials.cs:82:45:82:113 | "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+" | HardcodedCredentials.cs:82:45:82:113 | "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+" | The hard - coded value "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+" flows to the $@ in $@. | HardcodedCredentials.cs:82:45:82:113 | "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+" | setter call argument | .. / .. / .. / resources / stubs / _frameworks / Microsoft.AspNetCore.App / Microsoft.Extensions.Identity.Core.cs:667:31:667:55 | AllowedUserNameCharacters | AllowedUserNameCharacters |
+| HardcodedCredentials.cs:89:33:89:42 | "username" | HardcodedCredentials.cs:89:33:89:42 | "username" | HardcodedCredentials.cs:89:33:89:42 | "username" | The hard - coded value "username" flows to the $@ in $@. | HardcodedCredentials.cs:89:33:89:42 | "username" | setter call argument | .. / .. / .. / resources / stubs / _frameworks / Microsoft.AspNetCore.App / Microsoft.Extensions.Identity.Core.cs:25:31:25:47 | UserNameClaimType | UserNameClaimType |
 | TestHardcodedCredentials.cs:26:19:26:28 | "username" | TestHardcodedCredentials.cs:26:19:26:28 | "username" | TestHardcodedCredentials.cs:26:19:26:28 | "username" | The hard - coded value "username" flows to the $@ parameter in $@. | TestHardcodedCredentials.cs:26:19:26:28 | "username" | name | TestHardcodedCredentials.cs:24:31:38:13 | object creation of type MembershipUser | object creation of type MembershipUser |
```

</p>
</details>

### CODE REVIEW

File: csharp/ql/test/query-tests/Security Features/CWE-798/options [hunk 1]
```diff
@@ -1,4 +1,10 @@
 semmle-extractor-options: /nostdlib /noconfig
 semmle-extractor-options: --load-sources-from-project:${testdir}/../../../resources/stubs/System.Data.SqlClient/4.8.3/System.Data.SqlClient.csproj
 semmle-extractor-options: ${testdir}/../../../resources/stubs/System.Web.cs
+semmle-extractor-options: ${testdir}/../../../resources/stubs/_frameworks/Microsoft.AspNetCore.App/Microsoft.Extensions.Primitives.cs
```
Comment (michaelnebel) [inline csharp/ql/test/query-tests/Security Features/CWE-798/options:4]: Replace all the `AspNet` core extractor options with
semmle-extractor-options: --load-sources-from-project:${testdir}/../../../resources/stubs/_frameworks/Microsoft.AspNetCore.App/Microsoft.AspNetCore.App.csproj

**Review by michaelnebel — COMMENTED** [review]
This looks really good @felickz !
I will appreciate if you just make a minor change to the test extractor options.
Also, I will just start a DCA run to see if it has any impacts on the performance and/or results.

File: csharp/ql/test/query-tests/Security Features/CWE-798/options (hunk 1 above)
Comment (felickz) [reply csharp/ql/test/query-tests/Security Features/CWE-798/options]: Good observation, i just included one by one until compiler errors went away 😅

**Comment by michaelnebel** [comment]
DCA looks good (both in terms of results and performance).

**Review by michaelnebel — APPROVED** [review]
LGTM 👍 
Thank you for doing this @felickz !

**Pull request merged**